src/garage/sampler/on_policy_vectorized_sampler.py

```diff
@@ -103,27 +103,31 @@ def obtain_samples(self, itr, batch_size=None, whole_paths=True):
 
         policy = self.algo.policy
 
+        agent_infos = policy.get_initial_state(dones)
```

Comment:
refresh my memory on why `dones` is a necessary part of this API?

Reply:
so that policy can obtain the number of parallel environments. We can actually explicitly pass that rather than dones.

Reply:
see my other comments in this PR. i don't think `dones` is the right way to do this.